airflow/operators/python_operator.py

```diff
@@ -132,10 +132,8 @@ class BranchPythonOperator(PythonOperator, SkipMixin):
     to be inferred.
 
     Note that using tasks with ``depends_on_past=True`` downstream from
-    ``BranchPythonOperator`` is logically unsound as ``skipped`` status
-    will invariably lead to block tasks that depend on their past successes.
-    ``skipped`` states propagates where all directly upstream tasks are
-    ``skipped``.
+    ``BranchPythonOperator`` is fine because depends_on_past allows for
+    past tasks to be skipped.
```

Comment:
Lets just remove the paragraph entirely if this is the case? (Since when was this the case?)

Reply:
Done, sorry for the delay.

The behavior changed initially in https://github.com/apache/airflow/commit/2630361ca24737c28f458825b20ab11c9c996b17, leading to https://github.com/apache/airflow/commit/f360414774f1ecb2e0f4e53ebdd623c7435f9a78#diff-a32a363fa616685db3bfefba947535b2L1145, and is now true based on https://github.com/apache/airflow/blob/de75b7a2bd7f5bef6a1d09942e0b43c17a3fbb95/airflow/ti_deps/deps/prev_dagrun_dep.py#L75